Source/Dafny/AST/DafnyAst.cs

```diff
@@ -13,14 +13,92 @@
 using System.Diagnostics.Contracts;
 using System.Numerics;
 using System.Linq;
-using Microsoft.Boogie;
 using System.Diagnostics;
 using System.Threading;
+using Microsoft.Boogie;
 
 namespace Microsoft.Dafny {
   [System.AttributeUsage(System.AttributeTargets.Field)]
   public class FilledInDuringResolutionAttribute : System.Attribute { }
 
+  public interface IToken : Microsoft.Boogie.IToken {
```

Comment:
I would prefer not to inherit from Boogie tokens.

Reply:
The reason I agree down the road to not inherit Boogie tokens will be when we will package the verification through Boogie as a different dependency, so we will want to break the dependency.
However, this is not in the near future, Boogie is still be a dependency for now as far as I can tell. Moreover, when we translate the Dafny AST to Boogie, we need to provide Boogie tokens. Inheriting from Boogie tokens ensures we don't have the need to convert Dafny tokens to Boogie tokens, which is redundant, costly in time and resources (there are a lot of tokens) and thus not necessary for now.

I made the refactoring so that, at least, when we will want to break up 💔  with Boogie, imports of Boogie's tokens are already gone, and we will just need a few conversions there and there, which is a boring refactoring.

Reply:
I looked into how easy it would be to have a Dafny token that doesn't inherit form the Boogie one, as a separate PR, and found that because C# implicit conversions are only between classes, so not between `Dafny.Token` and `Boogie.IToken`, it takes lots of tiny changes in the translator code to do those casts explicitly.

So that refactoring is a little expensive for what it brings and I think it's OK to leave it for now.